src/layers/legacy/medVtkInria/vtkImageView/vtkImageViewCornerAnnotation.cxx

```diff
@@ -170,427 +170,183 @@ void vtkImageViewCornerAnnotation::TextReplace(vtkImageActor *ia,
         }
     }
 
-    // search for tokens, replace and then assign to TextMappers
-    for (i = 0; i < 4; i++)
+    int isKnownAnnotationStyle = -1;
+    if (view2d)
     {
-        if (this->CornerText[i] && strlen(this->CornerText[i]))
-        {
-            size_t len = strlen(this->CornerText[i]) + 1000;
-            text  = new char[len];
-            text2 = new char[len];
-            strncpy(text, this->CornerText[i], strlen(this->CornerText[i])+1);
-
-            // now do the replacements
+        isKnownAnnotationStyle = static_cast<int>(view2d->GetAnnotationStyle2D());
+    }
 
-            rpos = strstr(text, "<image>");
-            while (rpos)
-            {
-                *rpos = '\0';
-                if (ia && this->ShowSliceAndImage)
-                {
-                    snprintf(text2, len, "%sImage: %i%s", text, slice, rpos+7);
-                }
-                else
-                {
-                    snprintf(text2, len, "%s%s", text, rpos+7);
-                }
-                tmp = text;
-                text = text2;
-                text2 = tmp;
-                rpos = strstr(text, "<image>");
-            }
+    // Search for tokens, replace and then assign to TextMappers
+    QString replaceText = "";
+    QString textQ = "";
 
-            rpos = strstr(text, "<image_and_max>");
-            while (rpos)
-            {
-                *rpos = '\0';
-                if (ia && this->ShowSliceAndImage)
-                {
-                    snprintf(text2, len, "%sImage: %i / %i%s", text, slice, slice_max, rpos+15);
-                }
-                else
-                {
-                    snprintf(text2, len, "%s%s", text, rpos+15);
-                }
-                tmp = text;
-                text = text2;
-                text2 = tmp;
-                rpos = strstr(text, "<image_and_max>");
-            }
+    if (ia && this->ShowSliceAndImage && this->ImageView)
+    {
+        //---- osSW
```

Comment:
in the original code, we tried every tag (<value>, <zoom>, etc) to the text of each corner (there are four corners). The corner tags are completely different.
Here, we search only the good tags for each corner.